AppCenter/AppCenter/Internals/Context/Device/MSDeviceTracker.m

```diff
@@ -250,13 +250,20 @@ - (MSDevice *)deviceForTimestamp:(NSDate *)timestamp {
 
 - (void)clearDevices {
   @synchronized(self) {
+      
+    // Remember the information about the current device.
+    MSDeviceHistoryInfo *currentDeviceHistory = [[MSDeviceHistoryInfo alloc] initWithTimestamp:[NSDate date]
```

Comment:
I guess the history already contains current device info. Can we just remove all except the latest one?